docs/vaahcms-2/backend/permissions.md

```diff
@@ -0,0 +1,266 @@
+
+| Permissions          | Description                                           |
+| ------------- | ------------------------------------------------------------ |
+| Has Access Of Permissions Section | This will enable users to access the permission section link in the application's left navigation bar. |
+| Can Read Permissions | This will allow user to view anything in the permission section. |
+| Can Update Permissions | This will allow user to edit anything in the permission section. |
+| Can Manage Permissions | This will allow user to manage anything in the permission section. |
+
+## Files
+
+- Laravel Route: `vaahcms/Routes/backend/route-permissions.php`
+- Laravel Controller: `vaahcms/Http/Controllers/Backend/PermissionsController.php`
+- Laravel Model: `vaahcms/Models/Permission.php`
+- Vue Route: `vaahcms/Vue/routes/vue-routes-permissions.js`
+- Vue Store: `vaahcms/Vue/stores/store-permissions.js`
+- Vue Page Directory: `vaahcms/Vue/pages/permissions`
+
+## Methods
+
+Some reusable methods mention bellowed.
+
+#### Permission::getActiveItems()
+
+You can use this method for fetching all active permissions of the application.
+
+```php 
+use WebReinvent\VaahCms\Models\Permission;    // Import permission class at top
+
+$active_permissions = Permission::getActiveItems();
+```
+---
+
+#### Permission::syncPermissionsWithRoles()
+
+You can use this method for sync permissions with roles. After creating a new permission you have to call this method to sync permissions count with roles.
+
+```php
+use WebReinvent\VaahCms\Models\Permission;  // Import Role class at the top
+
+Permission::syncPermissionsWithRoles(); 
+```
+---
+
+#### Permission::getPermissionRoles($id)
+
+You can use this method to retrieve roles that are associated with this permission.
+
+```php
+use WebReinvent\VaahCms\Models\Permission;  // Import Role class at the top
+
+Permission::getPermissionRoles($id);        // Permission ID must be provided as parameters.
+```
+
+## API
+
+VaahCMS has APIs for every method, allowing you to interact with `NuxJS` or other frameworks.
+
+You can access APIs method from `vaachms/Http/Controllers/Api/PermissionsController.php` and routes from `vaahcms/Routes/api/api-routes-permissions.php`
+
+We mention some methods bellow which help you to understand the structure.
+
+#### Fetch Permissions
+
+##### Method: `GET`
+##### URL: `<public-url>/api/vaah/permissions/`
+
+##### Sample Request
+```php 
+parameter = [
+    'filter' => [
+        'q'            // for search queary (optionl)
+        'trashed'      // for include or exclude trashed data (optionl)
+        'is_active'    // for fetching only active or inactive data (optionl)
+        'sort'         // for sorting the data (optionl)
+    ];  
+];
+```
+
+##### Sample Response
+
+```json
+{
+  "data": {
+      .............
+      .............
+  },
+  "success": "true"
+}
+```
+---
+
+#### Delete Permission
+
+###### Method: `DELETE`
+###### Action: `deleteItem`
```

Comment:
@wri-tanmoy-d001 what is this `deleteItem`? is it from Vue or Model?

Reply:
@themodernpk sir,

Sir, I'll update the PR
and sir `deleteItem` method part of the laravel model